CIPs/cip-granda_mento.md

```diff
@@ -0,0 +1,138 @@
+
+A new contract, `GrandaMento.sol`, is created. One deployment of `GrandaMento` exists per stable token. It must have the ability to mint/burn the corresponding stable token and be a spender of `Reserve.sol`.
+
+The contract has the following configurable parameters:
+
+1. **Approver address** - Set by governance. Intended to be a multisig and has the authority to approve trades.
+2. **Minimum stable token buy/sell quantity for trades** - Set by governance. Minimum amount of stable token a trade must buy/sell.
+3. **Required waiting period for an approved trade before execution** - Set by governance. The minimum amount of time that must elapse between a proposed trade being approved and when the trade can be executed. Should give sufficient time for Governance to veto an approved trade.
+4. **Spread imposed on trades** - Set by governance. The percent fee imposed upon the trade.
+
+The contract has the following functions:
+
+1. **Propose trade** - Called by a trader to initiate a trade.
+   * Callable by anyone.
+   * Requires the trade to be larger than the minimum stable token buy/sell quantity.
+   * Deposits the full amount of the asset being sold into the contract.
+   * Records:
+     * Which asset is being sold
+     * The amount of the asset being sold
+2. **Approve trade proposal** - Approves a proposed trade.
+   * Only callable by the approver address.
+   * Marks a proposed trade as approved.
+   * Records:
+     * The timestamp at which the approval for the proposed trade occurred.
+3. **Veto an approved trade proposal** - Vetoes a previously approved proposed trade.
+   * Only callable by governance.
+   * Refunds the proposed trade's deposited sell asset.
+   * Records:
+     * The trade as vetoed.
+4. **Execute an approved and ready trade** - Executes a trade.
+   * Callable by anyone.
+   * Requires the trade to have been approved.
+   * Requires the required waiting period for an approved trade to have elapsed since the time it was approved.
+   * Makes the trade:
+     * If selling CELO and purchasing stable token:
+       * The CELO is sent to the Reserve.
+       * Stable token is minted to the trader according to the rate originally recorded when the trade was proposed.
+     * If selling stable token and purchasing CELO:
+       * The stable token is burned.
+       * CELO is transferred from the Reserve to the trader according to the rate originally recorded when the trade was proposed.
+   * Records:
+     * The trade as executed.
+
+#### Modifications of existing contracts:
+
+Currently, `StableToken.sol` only allows its `Exchange.sol` or `Validators.sol` to [mint](https://github.com/celo-org/celo-monorepo/blob/master/packages/protocol/contracts/stability/StableToken.sol#L225) new stable tokens. Similarly, only its `Exchange.sol` can [burn](https://github.com/celo-org/celo-monorepo/blob/master/packages/protocol/contracts/stability/StableToken.sol#L275) stable tokens.
+
+The following modifications will be made:
```

Comment:
An alternative implementation could have the "mint/burn" limit be set in Granda Mento by Governance.